aws/data_source_aws_ec2_transit_gateway_dx_gateway_attachment.go

```diff
@@ -18,35 +18,56 @@ func dataSourceAwsEc2TransitGatewayDxGatewayAttachment() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"dx_gateway_id": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 			},
 			"tags": tagsSchemaComputed(),
 			"transit_gateway_id": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 			},
+			"filter": dataSourceFiltersSchema(),
 		},
 	}
 }
 
 func dataSourceAwsEc2TransitGatewayDxGatewayAttachmentRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
+	filters, filtersOk := d.GetOk("filter")
+	tags, tagsOk := d.GetOk("tags")
+	dxGatewayId, dxGatewayIdOk := d.GetOk("dx_gateway_id")
+	transitGatewayId, transitGatewayIdOk := d.GetOk("transit_gateway_id")
+
 	input := &ec2.DescribeTransitGatewayAttachmentsInput{
 		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("resource-type"),
+				Values: []*string{aws.String(ec2.TransitGatewayAttachmentResourceTypeDirectConnectGateway)},
+			},
+		},
+	}
+	if filtersOk {
+		input.Filters = append(input.Filters, buildAwsDataSourceFilters(filters.(*schema.Set))...)
+	}
+	if tagsOk {
+		input.Filters = append(input.Filters, ec2TagFiltersFromMap(tags.(map[string]interface{}))...)
+	}
+	// to preserve original functionality
+	if dxGatewayIdOk && transitGatewayIdOk {
+		input.Filters = []*ec2.Filter{
 			{
 				Name:   aws.String("resource-id"),
-				Values: []*string{aws.String(d.Get("dx_gateway_id").(string))},
+				Values: []*string{aws.String(dxGatewayId.(string))},
 			},
 			{
 				Name:   aws.String("resource-type"),
-				Values: []*string{aws.String("direct-connect-gateway")}, // Not yet defined in ec2/api.go.
+				Values: []*string{aws.String(ec2.TransitGatewayAttachmentResourceTypeDirectConnectGateway)},
 			},
 			{
 				Name:   aws.String("transit-gateway-id"),
-				Values: []*string{aws.String(d.Get("transit_gateway_id").(string))},
+				Values: []*string{aws.String(transitGatewayId.(string))},
 			},
-		},
+		}
 	}
```

Comment:
Same here as the other PR. 👍 (removing `resource-type` so its not duplicated in the request)

```go
	if dxGatewayIdOk {
		input.Filters = append(input.Filters, &ec2.Filter{
			Name:   aws.String("resource-id"),
			Values: []*string{aws.String(dxGatewayId.(string))},
		})
	}

	if transitGatewayIdOk {
		input.Filters = append(input.Filters, &ec2.Filter{
			Name:   aws.String("transit-gateway-id"),
			Values: []*string{aws.String(transitGatewayId.(string))},
		})
	}
```

Reply:
done + small changes to naming. 

--- PASS: TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_TransitGatewayIdAndDxGatewayId (1188.58s)
--- PASS: TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_filter (1256.65s)